src/handlers/ExecuteActionButtonHandler.ts

```diff
@@ -26,11 +38,167 @@ export class ExecuteActionButtonHandler {
-        const { actionId, user, room, triggerId, message } = this.context.getInteractionData();
+        const { actionId, user, room, triggerId } = this.context.getInteractionData();
 
-        // Handle any future action button interactions here
-        // Currently we don't have any action buttons, but this is prepared for future use
+        try {
+            switch (actionId) {
+                case ActionIds.SEND_EMAIL_DIRECT_ACTION:
+                    await this.handleDirectSendEmail(user, room);
+                    break;
+                    
+                case ActionIds.SEND_EMAIL_EDIT_ACTION:
+                    await this.handleEditAndSendEmail(user, room, triggerId);
+                    break;
+                    
+                default:
+                    break;
+            }
+        } catch (error) {
+            // Log error but don't throw to avoid breaking the interaction
+            this.app.getLogger().error('Error handling action button:', error);
+        }
 
         return this.context.getInteractionResponder().successResponse();
     }
+
+    private async handleDirectSendEmail(user: any, room: any): Promise<void> {
+        const language = await getUserPreferredLanguage(
+            this.read.getPersistenceReader(),
+            this.persistence,
+            user.id,
+        );
+
+        try {
+            // Retrieve stored email data
+            const emailData = await this.getStoredEmailData(room.id);
+            if (!emailData) {
+                throw new Error('Email data not found');
+            }
+
+            // Send email directly using ToolExecutorService
+            const toolExecutorService = new ToolExecutorService(
+                this.app,
+                this.read,
+                this.modify,
+                this.http,
+                this.persistence
+            );
+
+            const result = await toolExecutorService.sendEmail(emailData, user);
+
+            // Show result message
+            await this.showMessage(
+                user,
+                room,
+                result.success ? `${result.message}` : `${result.message}`,
+                language
+            );
+
+        } catch (error) {
+            // Show error message
+            await this.showMessage(
+                user,
+                room,
+                `${t(Translations.SEND_EMAIL_FAILED, language, { error: 'Unknown error' })}`,
+                language
+            );
+        }
+    }
+
+    private async handleEditAndSendEmail(user: any, room: any, triggerId?: string): Promise<void> {
+        const language = await getUserPreferredLanguage(
+            this.read.getPersistenceReader(),
+            this.persistence,
+            user.id,
+        );
+
+        try {
+            // Retrieve stored email data
+            const emailData = await this.getStoredEmailData(room.id);
+            
+            if (!emailData) {
```

Comment:
User should get a error message when these error's occur  that explain what went wrong.